### PR TITLE
Add editor example

### PR DESCRIPTION
**Pull request by nikku**
This contributes the editor example previously found in diagram-js/example.

Not only did we add the example, but also improve it (i.e. showcase the editor vs app separation...).

---

Related to https://github.com/bpmn-io/diagram-js/issues/332

### CODE REVIEW

File: editor/src/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import Editor from './editor';
+
+console.log(Editor);
```
Comment (pinussilvestrus) [inline editor/src/index.js]: is this intended?

**Review by pinussilvestrus — COMMENTED** [review]
Looks good to me 👍 only one single comment.

File: editor/src/index.js (hunk 1 above)
Comment (nikku) [reply editor/src/index.js]: Nope. :wink:

Comment (nikku) [reply editor/src/index.js]: Fixed.

**Pull request merged**